generator/src/main/java/com/scottlogic/deg/generator/inputs/validation/ProfileValidationVisitor.java

```diff
@@ -0,0 +1,195 @@
+package com.scottlogic.deg.generator.inputs.validation;
+
+import com.scottlogic.deg.generator.ProfileFields;
+import com.scottlogic.deg.generator.Rule;
+import com.scottlogic.deg.generator.constraints.Constraint;
+import com.scottlogic.deg.generator.constraints.atomic.*;
+import com.scottlogic.deg.generator.inputs.validation.validators.*;
+import com.scottlogic.deg.generator.restrictions.Nullness;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class ProfileValidationVisitor implements ProfileVisitor {
+
+    private Map<String, ConstraintValidator> allFieldsState;
```

Comment:
You can hold a `Map<Field, ConstraintValidator>` as `Field` implements `hashCode()` and `equals()`

Reply:
done.